bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryDirectDataWriterHelper.java

```diff
@@ -252,13 +252,7 @@ private void checkForFailedResponse(boolean waitForResponse) {
    *     (deduplication error) or if the response contains an error.
    */
   private void sendAppendRowsRequest() throws IOException {
-    boolean waitForResponse = false;
-    if (writeAtLeastOnce) {
-      // FUTURE: Enable async writes with default stream once client retries are supported.
-      // When using default stream we may encounter retry-able server errors when under high load.
-      waitForResponse = true; // disable async writes
-    }
-    checkForFailedResponse(waitForResponse);
+    checkForFailedResponse(false /*waitForResponse*/);
```

Comment:
should we make changes in finalizeStream as well?
```
    if (!this.writeAtLeastOnce) {
      waitBeforeFinalization();
```

Reply:
No. No changes are needed in finalizeStream(). There, we first call checkForFailedResponse(true) to wait for all outstanding responses to be received, to ensure there were no errors. Then, if we are using the default stream there is no need to call Finalize() on it.